internal/painter/gl/gl_common.go

```diff
@@ -81,16 +81,55 @@ func (p *glPainter) newGlLineTexture(obj fyne.CanvasObject) Texture {
 }
 
 func (p *glPainter) newGlRectTexture(rect fyne.CanvasObject) Texture {
+	if rect, ok := rect.(*canvas.Rectangle); ok {
```

Comment:
Complexity here is increased a lot - it's basically a new function inline.
Hopw about: going back to the previous type check, and if it's got no stroke then using the previous code (much faster to render!). Then if there is a stroke calling out to a new method newGlStrokedRectTexture()?
I think that would be easier to read.

Reply:
Thanks for the review!
Is it okay like this?

```go
func (p *glPainter) newGlRectTexture(rect fyne.CanvasObject) Texture {
	col := theme.BackgroundColor()
	if wid, ok := rect.(fyne.Widget); ok {
		widCol := cache.Renderer(wid).BackgroundColor()
		if widCol != nil {
			col = widCol
		}
	} else if rect, ok := rect.(*canvas.Rectangle); ok {
		// add if block like this
		if rect.Stroke {
			return newGlStrokedRectTexture()
		}
		if rect.FillColor != nil {
			col = rect.FillColor
		}
	}

	return p.imgToTexture(image.NewUniform(col), canvas.ImageScaleSmooth)
}
```

Reply:
Yeah I think that looks quite neat :)